src/styles/form-message.scss

```diff
@@ -35,17 +49,37 @@ $block: #{$fd-namespace}-form-message;
 
   &--error {
     background-color: var(--sapErrorBackground);
+
+    &::before {
+      content: "\e0b1";
+      color: var(--sapNegativeElementColor);
+    }
   }
 
   &--warning {
     background-color: var(--sapWarningBackground);
+
+    &::before {
+      content: "\e201";
+      color: var(--sapCriticalElementColor);
+    }
   }
 
   &--success {
     background-color: var(--sapSuccessBackground);
+
+    &::before {
+      content: "\e203";
+      color: var(--sapPositiveElementColor);
+    }
   }
 
   &--information {
     background-color: var(--sapInformationBackground);
+
+    &::before {
+      content: "\e202";
+      color: var(--sapInformativeElementColor);
+    }
```

Comment:
Lots of repeating code. Could you make a mixin that sets the state. You can pass the background-color, content and color as params. 
Another option could be to put all these values in an object and loop through it, like we do for many other components.